### PR TITLE
Add a github issue reference (and a cache manager)

### LINKED ISSUE #25: Display an embed of the GitHub issue when including a GitHub issue ID in your message
### Feature request

**Checklist**
- [x] I have read and agree to the README (found on https://github.com/Minehut/Meta)
- [x] This is not a bug report, support request or plugin request
- [x] I have searched the Issue Tracker to make sure this is not a duplicate issue
- [x] This issue does not contain any sensitive information 

**Platform**

Discord (Meta)

**Describe the feature**

Display an embed of the GitHub issue when including a GitHub issue ID in your message.
The issue ID, title, and an excerpt from the issue description should be included.
Example:
```
CoolUser: Have you seen this issue? #1234
MH Meta:
┍━━━━━━━━━━━━━┑
╿#1234 - Blah.╿
╿Excerpt...   ╿
┕━━━━━━━━━━━━━┙
```

**How the feature is useful**

It will make linking GitHub issues easier when discussing them in the Minehut Meta Discord server.

---
**Use 👍 and 👎 reactions to indicate you agree or disagree with this suggestion! Feel free to discuss the idea in the comments.**

### PR DESCRIPTION
**Pull request by MiginsMC**
Closes [#25](https://github.com/Minehut/Meta/issues/25)

- When you type #(number) it sends an embed containing a truncated version of that github issue. It removes checklists and random whitespace to make the embed as informative as possible.  
- implements a cache feature which saves the issue information locally for 10 minutes (deletes after that time) to limit the number of API requests the bot is doing.
- implements a cooldown with the already existing cooldown manager to prevent spam, etc.
- guild config for ignored channels + repo info.

Hopefully, I didn't butcher this... but again thanks to @jellz for helping :)

### CODE REVIEW

File: src/listener/githubIssueReference.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+		const messageMatch = msg.content.match(/#\d+/);
+
+		if (messageMatch) {
+			const issueNumber = parseInt(messageMatch[0].slice(1));
+			if (
+				this.client.githubCooldownManager.isOnCooldown(
+					`gh-${issueNumber}-${msg.channel.id}`
+				)
+			)
+				return msg.react('⏲️');
```
Comment (theminecoder) [inline src/listener/githubIssueReference.ts]: Check the spacing here

File: src/structure/cacheManager.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import { Collection } from 'discord.js';
+
+const EXPIRED_MS = 8.64e7; // 24 hours
```
Comment (theminecoder) [inline src/structure/cacheManager.ts]: Let's make this 10 mins to allow for when descriptions/labels are updated

**Review by theminecoder — CHANGES_REQUESTED** [review]

File: src/structure/cacheManager.ts (hunk 2 above)
Comment (ckiee) [reply src/structure/cacheManager.ts]: 10 mins = `6e5` ms

File: src/listener/githubIssueReference.ts (hunk 1 above)
Comment (MiginsMC) [reply src/listener/githubIssueReference.ts]: Weird, prettier likes it indented there, 🤷

File: src/guild/config/guildConfigs.ts [hunk 3]
```diff
@@ -172,7 +172,11 @@ guildConfigs.set('546414872196415501', {
 		hastebinConversion: {
 			ignoredChannels: ['548317804076597249'],
 			whitelistedExtensions: HASTEBIN_EXTENSIONS_WHITELIST
-		}
+    	},
```
Comment (theminecoder) [inline src/guild/config/guildConfigs.ts]:
```suggestion
   },
```

File: src/structure/cacheManager.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
+import { Collection } from 'discord.js';
+
+const EXPIRED_MS = 6e5; // 10 minutes
```
Comment (theminecoder) [inline src/structure/cacheManager.ts]:
```suggestion
const EXPIRED_MS = 600000; // 10 minutes
```

File: src/guild/config/guildConfigs.ts [hunk 5]
```diff
@@ -172,7 +172,11 @@ guildConfigs.set('546414872196415501', {
 		hastebinConversion: {
 			ignoredChannels: ['548317804076597249'],
 			whitelistedExtensions: HASTEBIN_EXTENSIONS_WHITELIST
-		}
+   },
```
Comment (theminecoder) [inline src/guild/config/guildConfigs.ts]:
```suggestion
    },
```

**Review by theminecoder — APPROVED** [review]

**Comment by MiginsMC** [comment]
should there be any ignored channels in the meta discord? Probably don't want ppl using that in #meta-feedback and #filter-requests.

**Comment by Sniped** [comment]
probably logs too

**Pull request merged**